client/hwlib/src/collectors/os_info.rs

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+use anyhow::Result;
+use regex::Regex;
+use std::fs;
+use std::process::Command;
+
+use crate::models::software;
+
+pub fn collect_os_info(proc_version_filepath: &'static str) -> Result<software::OS> {
+    let codename = super::os_info::get_codename()?;
+    let distributor = super::os_info::get_distributor()?;
+    let version = super::os_info::get_version()?;
+    let kernel = collect_kernel_info(proc_version_filepath)?;
+
+    let os_info = software::OS {
+        codename,
+        distributor,
+        version,
+        kernel,
+    };
+
+    Ok(os_info)
+}
+
+pub fn collect_kernel_info(proc_version_filepath: &'static str) -> Result<software::KernelPackage> {
+    let kernel_version = fs::read_to_string(proc_version_filepath)?;
+    let kernel_version = kernel_version
+        .split_whitespace()
+        .nth(2)
+        .unwrap_or("")
```

Comment:
I think this can be `.unwrap_or_default()`

Reply:
Updated in fd7e3cb